src/duk_error_misc.c

```diff
@@ -5,6 +5,35 @@
 #include "duk_internal.h"
 
 /*
+ * Helper to walk the thread chain and see if there is an active error
+ * catcher. Protected calls aren't considered.
+ */
+
+DUK_LOCAL duk_bool_t duk__have_active_catcher(duk_hthread *thr) {
+	/*
+	 * XXX: As noted above, a protected API call won't be counted as a
+	 * catcher. This is usually convenient, e.g. in the case of a top-
+	 * level duk_pcall(), but may not always be desirable. Perhaps add an
+	 * argument to treat them as catchers?
+	 */
+
+	duk_size_t i;
+
+	DUK_ASSERT(thr != NULL);
+
+	while (thr != NULL) {
+		for (i = 0; i < thr->catchstack_top; i++) {
+			duk_catcher *cat = thr->catchstack + i;
+			if (DUK_CAT_HAS_CATCH_ENABLED(cat)) {
+				return 1;  /* all we need to know */
```

Comment:
I was going to do that, but then information on the original throw site is lost if it IS rethrown (execution pauses at ENDFIN), which is the common case in a finally with no catch.  By that point the stack is already unwound so that pause is useless.